### PR TITLE
use autosummary instead of autodoc for API reference

### PR DESCRIPTION
**Pull request by MRossol**
@daniel-thom I spent all day figuring out how to properly use sphinx autosummary to produce clean API reference docs. I've implemented it here if you want to give it a go

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -20,11 +20,11 @@ help:
 
 %: Makefile
 	python make_model_tables.py -o $(BUILDDIR)/model_tables
+	rm -rf build
+	rm -rf source/_autosummary
```
Comment (daniel-thom) [inline docs/Makefile:24]: Won't this break incremental builds? Should this only occur in `make clean`?

I already have something kinda hacky here. The python script on line 22 builds some tables into `build` that I include in the docs. It needs to be defined properly with a make target (it runs even when you call `make clean`). If we really need to delete these directories then the python script needs to get called afterwards.

**Comment by daniel-thom** [comment]
> @daniel-thom I spent all day figuring out how to properly use sphinx autosummary to produce clean API reference docs. I've implemented it here if you want to give it a go

Thanks for the PR. I'm interested to see what this produces. Does it change the API docs or is this really for referencing the APIs within the documentation pages?

File: docs/Makefile (hunk 1 above)
Comment (MRossol) [reply docs/Makefile:24]: woops, I meant to put this above the build tables. I'll push that now. This just ensures you get a clean build everytime you make the docs

**Comment by MRossol** [comment]
Autosummary IMO creates more user friendly docs for the API and the changes to index put them in the side bar under API reference. See the rex example here:
https://nrel.github.io/rex/_autosummary/rex.html

**Review by daniel-thom — APPROVED** [review]

**Pull request merged**